hutch_python/plugins/happi.py

```diff
@@ -0,0 +1,51 @@
+"""
+This plugin depends on a JSON happi database stored somewhere in the file
+system. The path of this file is supplied in the YAML as ``filename``.
+Finally, if a smaller subsection of the databse is desired, a requirements
+specification can be used to limit the number of devices that are loaded.
+
+The plugin then handles; initalizing the ``happi.Client``, finding the
+containers that match the specified requirements, then using the device
+loading utilities from happi to create instantiated devices
+
+Example
+-------
+.. code:: YAML
+
+    happi:
+        filename: path/to/my_db.json
+        requirements:
+            active: True
+            beamline: MFX
+"""
+import logging
+
+import happi
+from happi.loader import load_devices
+
+from ..base_plugin import BasePlugin
+
+logger = logging.getLogger(__name__)
+
+
+class Plugin(BasePlugin):
+    """
+    Plugin to load search information from happi
+    """
+    priority = 5
```

Comment:
No need to assign a priority unless it's really important that your plugin runs first/last or at a specific point in time

Reply:
Actually, to expand a bit, assigning a priority when it isn't needed can cause confusion.

Reply:
Still developing thoughts on this... The appropriate activate timing for this plugin is probably "before `lightpath`". So the priority should come before the default priority and after the namespace priority, so I think 5 is good.

I'm considering ditching this system and moving to a more fluid cause-and-effect system where you can specify which plugins need to be run before this one can be run.

Reply:
I'll come back for this when I do the `lightpath` plugin